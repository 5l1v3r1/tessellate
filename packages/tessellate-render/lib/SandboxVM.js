// @flow

import vm from 'vm'

function isExport(thing: ?mixed): boolean {
  return typeof thing === 'object' || typeof thing === 'function'
}

export default class SandboxVM {
  globals: Object;

  constructor(globals: Object = {}) {
    this.globals = globals
  }

  run(source: string): Object {
    const globals = {
      module: {},
      exports: {}
    }
    globals.module.exports = globals.exports
    const script = new vm.Script(source, {displayErrors: true})
    const sandbox = Object.assign({}, globals, this.globals)
    const context = vm.createContext(sandbox)
    const result = script.runInContext(context)

    if (result && isExport(result.default))
      return result.default
    if (isExport(result))
      return result
    if (isExport(sandbox.module.exports))
      return sandbox.module.exports
    if (isExport(sandbox.exports))
      return sandbox.exports
    else
      return {}
  }
}
