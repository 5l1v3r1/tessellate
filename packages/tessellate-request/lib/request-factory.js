// @flow

import JsonFileBackedOAuthRequest from './requests/file-backed-oauth-request'
import DefaultRequest from './requests/default-request'
import type {SmartRequest} from './requests/smart-request'
import * as configurationProvider from './configuration/configuration-provider'

const OAUTH2_JSON_FILE_BACKED_USER_CREDENTIALS_PROVIDER = 'OAUTH2_JSON_FILE_BACKED_USER_CREDENTIALS_PROVIDER'

export function createAuthRequestById(id: string): SmartRequest {

  const config: AuthConfig = configurationProvider.getById(id)

  if(config.type === OAUTH2_JSON_FILE_BACKED_USER_CREDENTIALS_PROVIDER) {
    return new JsonFileBackedOAuthRequest(config)
  } else {
    throw new Error(`${config.type} authentication type is not supported`)
  }
}

export function createDefault(): SmartRequest {
  return new DefaultRequest()
}
