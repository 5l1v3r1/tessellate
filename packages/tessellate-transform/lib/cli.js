// @flow

import path from 'path'
import fs from 'mz/fs'
import yargs from 'yargs'
import parse from './'

import type { File } from './'
import type { ParseOptions } from './parsers'

type Options = {|
  file: File;
  parser: ParseOptions;
|};

async function parseArgs(args: Object): Promise<Options> {
  const filepath = args._[0]
  if (typeof filepath !== 'string') {
    throw new Error('Missing argument: input file')
  }
  const extname = path.extname(filepath)
  if (extname !== '.jsx' && extname !== '.json' && extname !== '.yaml' && extname !== '.xml') {
    throw new Error(`Unsupported file: ${filepath}`)
  }
  const buffer = await fs.readFile(filepath)

  return {
    file: {
      content: buffer.toString(),
      extname
    },
    parser: {
      typePrefix: args.typePrefix
    }
  }
}

async function main(args: Object): Promise<void> {
  const options = await parseArgs(args)
  const result = await parse(options.file, options.parser)
  console.log(JSON.stringify(result))
}

// $FlowIssue https://github.com/facebook/flow/issues/1362
if (require.main === module) {
  const argv = yargs
    .usage('tessellate-transform [OPTIONS] FILE')
    .string('type-prefix')
    .describe('type-prefix', 'Prefix for React class names.')
    .demand(1)
    .argv
  main(argv).catch(console.error)
}
