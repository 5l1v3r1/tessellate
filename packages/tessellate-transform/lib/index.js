// @flow

import parseJSX from './parsers/jsx-parser'

import type { ParseOptions } from './parsers'

export type File = {|
  content: string;
  extname: '.jsx' | '.json' | '.yaml' | '.xml';
|};

export default async function parse(file: File, opts: ParseOptions): Promise<*> {
  switch(file.extname) {
    case '.jsx':
      return parseJSX(file.content, opts)
    default:
      throw new Error(`Unsupported file type ${file.extname}`)
  }
}
