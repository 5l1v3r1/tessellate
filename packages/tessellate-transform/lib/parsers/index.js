// @flow

type JSONMap = {
  typeKeys: Array<string>;
  childrenKeys: Array<string>;
  literalKeys: Array<string>;
  ignoreKeys: Array<string>;
};

export type ParseOptions = {
  typePrefix?: string;
  jsonMap?: JSONMap;
};

type ParseResultObject = {
  type: string;
  props: Object | null;
  children: Array<ParseResultObject | ParseResultString>;
};

type ParseResultString = string;

export type ParseResult = ParseResultObject | ParseResultString;
