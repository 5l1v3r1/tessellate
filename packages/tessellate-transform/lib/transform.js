// @flow

import createJSXParser from './parsers/jsx-parser'

import type { Parser, ParseResult } from './parsers'

export default function transform(content: string, parse: Parser): * {
  let root: ParseResult | null = null
  const nodes = []
  const getLastNode = () => nodes[nodes.length - 1]

  const callbacks = {
    onEnter: (node: ParseResult) => {
      const lastNode = getLastNode()

      if (typeof lastNode === 'object') {
        lastNode.children.push(node)
      } else {
        root = node
      }
      nodes.push(node)
    },
    onLeave: (node: ParseResult) => {
      nodes.pop()
    },
    onLiteral: (literal: string) => {
      const lastNode = getLastNode()
      if (typeof lastNode === 'object') {
        lastNode.children.push(literal)
      }
    }
  }

  parse(content, callbacks)

  if(!root || typeof root !== 'object') throw new Error('Could not parse JSX.')
  else return root
}
