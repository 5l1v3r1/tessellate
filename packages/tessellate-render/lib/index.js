// @flow

import React from 'react';
import ReactDOMServer from 'react-dom/server';
import SandboxVM from './SandboxVM';

// Hack: global.__REQUIRE_FN__ must be defined in webpack.config.
const requireFn = global.__REQUIRE_FN__ || require;
const sandbox = new SandboxVM({ require: requireFn });

export default function renderToString(source: string, props: Object = {}): string {
  const { Fragment } = sandbox.run(source);
  return ReactDOMServer.renderToString(<Fragment {...props} />);
}
