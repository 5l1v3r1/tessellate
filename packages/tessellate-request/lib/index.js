// @flow

import type { SmartRequest } from './requests/smart-request';
import * as requestFactory from './requests/request-factory';

export default class Request {
  delegateRequest: SmartRequest;

  constructor(authId?: string) {
    if (authId) {
      this.delegateRequest = requestFactory.createAuthRequestById(authId);
    } else {
      this.delegateRequest = requestFactory.createDefault();
    }
  }

  async get(url: string): Object {
    return await this.delegateRequest.get(url);
  }
}
