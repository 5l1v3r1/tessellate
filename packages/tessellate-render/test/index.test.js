/**
 * @flow
 */

import * as renderService from '../lib'

describe('render-service', () => {

  // React component with UMD module definition (only supports CommonJS)
  const source = `
  (function (root, factory) {
    if (typeof exports === 'object' && typeof exports.nodeName !== 'string') {
      factory(exports, require('react'))
    }
  }(this, function (exports, React) {
    exports.Fragment = React.createClass({
      render: function() {
         return React.createElement('div', null,
          React.createElement('h1', null, 'Hello, world!'),
          React.createElement('p', null, this.props.text))
       }
    })
  }))
  `

  describe('renderToStaticMarkup', () => {
    it('should render a JavaScript component string to HTML', () => {
      const html = renderService.renderToStaticMarkup(source, {text: 'This is only a test.'})
      expect(html).toMatchSnapshot()
    })
  })
})
