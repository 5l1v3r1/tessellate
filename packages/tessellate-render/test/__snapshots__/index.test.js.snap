// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderToString should render a JavaScript component string to HTML 1`] = `"<div data-reactroot=\\"\\" data-reactid=\\"1\\" data-react-checksum=\\"-706139603\\"><h1 data-reactid=\\"2\\">Hello, world!</h1><p data-reactid=\\"3\\">This is only a test.</p></div>"`;
